src/backend/marsha/core/views.py

```diff
@@ -14,29 +14,69 @@
 from pylti.common import LTIException
 from rest_framework_simplejwt.tokens import AccessToken
 
-from .lti import LTI, OpenEdxLTI
+from .lti import LTI
+from .lti.utils import get_or_create_file, get_or_create_video
 from .models.account import INSTRUCTOR, STUDENT
-from .serializers import VideoSerializer
+from .serializers import FileSerializer, VideoSerializer
 from .utils.react_locales_utils import react_locale
 
 
 logger = getLogger(__name__)
 
+VIDEO, FILE = "videos", "files"
+SUPPORTED_RESOURCES = [VIDEO, FILE]
+RESOURCE_MAPPING = {VIDEO: "video", FILE: "file"}
+
 
 @method_decorator(csrf_exempt, name="dispatch")
 @method_decorator(xframe_options_exempt, name="dispatch")
-class VideoLTIView(TemplateResponseMixin, View):
+class LTIView(TemplateResponseMixin, View):
```

Comment:
This seems much too convoluted.

It would be simpler to share a BaseLTIView class and derive it into:

```python
class FileLTIView(BaseLTIView):
    model = File
    serializer_class = FileSerializer

class VideoLTIView(BaseLTIView):
    model = Video
    serializer_class = VideoSerializer
```

Reply:
:+1:

Reply:
But in that case we will don't have the generic route anymore:

```
path("lti/<resource:resource>/<uuid:uuid>", LTIView.as_view(), name="lti_view"),
```

we will have to duplicate it for every resources:

```
path("lti/videos/<uuid:uuid>", VideoLTIView.as_view(), name="lti_view"),
path("lti/documents/<uuid:uuid>", DocumentLTIView.as_view(), name="lti_view"),
```

Are you ok with this ?

Reply:
yes that's what I had in mind. It's more explicit to declare one route per resource as well. So :+1: